src/storage/store/node_column.cpp

```diff
@@ -143,7 +148,9 @@ void BoolNodeColumn::batchLookup(
 
 void NodeColumn::scan(
     Transaction* transaction, ValueVector* nodeIDVector, ValueVector* resultVector) {
-    nullColumn->scan(transaction, nodeIDVector, resultVector);
+    if (propertyStatistics.mayHaveNull()) {
+        nullColumn->scan(transaction, nodeIDVector, resultVector);
+    }
     scanInternal(transaction, nodeIDVector, resultVector);
 }
 
```

Comment:
There are two scan functions serving for different purposes, one used for string and var list (L157), and the other one for scanning a whole column chunk at a time (L171). Can we also apply the optimization to both interfaces?

Reply:
Right, I missed that. I think I'd just looked at the one which NullNodeColumn overrides and forgot to add it to the others.

I think it would probably work better if the check is instead done inside the NullNodeColumn::scan functions. That way the nullColumn can be used as usual and it will apply everywhere.